lib/components.js

```diff
@@ -11,20 +11,23 @@ export function SimpleStateProvider ({ initialState, reducers, middleware, child
   );
 }
 
-export function SimpleStateConsumer ({ mapState, mapDispatch, children }) {
+export function SimpleStateConsumer ({ mapState, mapDispatch, children, ConnectStateComponent = ConnectState }) {
   return createElement(
     StateContext.Consumer,
     null,
     ({ state, dispatch }) => createElement(
-      ConnectState,
+      ConnectStateComponent,
       { state, dispatch, mapState, mapDispatch },
       children
     )
   );
 }
 
-class ConnectState extends Component {
-  state = {};
+export class ConnectState extends Component {
+  constructor(props) {
```

Comment:
Can we remove this constructor please

Reply:
I'm calling super() in the ctor, so I don't think it can be removed (or will it be called automagically somehow?)

Reply:
It's just not necessary in this instance. What you're written is functionally identical to `state = {}`